src/main/java/gyro/azure/compute/VirtualMachineResource.java

```diff
@@ -807,6 +885,7 @@ public void create(GyroUI ui, State state) {
 
         setId(virtualMachine.id());
         setVmId(virtualMachine.vmId());
+        copyFrom(virtualMachine);
```

Comment:
Seems like the use of `copyFrom()` is a doing more than we expect. I understand that some of the fields cannot be known until the VM is actually launched. Will running this extra logic have any unintentional side effects?

Reply:
The call to copy from is especially needed in this case as the output fields (which are not configured by the user) are not saved in the state until we set it explicitly using the setter, after (create/update). The fields like `launchDate` and `state` would not be present when the ssh plugin calls the `gyro ssh` or `gyro list`

Reply:
Ok, my concern was that calling copyFrom() would have performance implications.